packages/ic-asset-certification/src/asset_config.rs

```diff
@@ -324,6 +403,97 @@ pub enum AssetRedirectKind {
     Temporary,
 }
 
+/// The encoding of an asset.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum AssetEncoding {
+    /// The asset is encoded with the Brotli algorithm.
+    Brotli,
+
+    /// The asset is encoded with the Zstd algorithm.
+    Zstd,
+
+    /// The asset is encoded with the Gzip algorithm.
+    Gzip,
+
+    /// The asset is encoded with the Deflate algorithm.
+    Deflate,
+}
+
+impl AssetEncoding {
+    /// Returns the default encoding and file extension for the encoding.
+    /// The default encoding is the encoding that is used when the client
+    /// does not specify an encoding in the `Accept-Encoding` header.
+    ///
+    /// The default encoding and file extension are:
+    /// - [Brotli](AssetEncoding::Brotli): `br`
+    /// - [Zstd](AssetEncoding::Zstd): `zst`
+    /// - [Gzip](AssetEncoding::Gzip): `gz`
+    /// - [Deflate](AssetEncoding::Deflate): `zz`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use ic_asset_certification::AssetEncoding;
+    ///
+    /// let (encoding, extension) = AssetEncoding::Brotli.default();
+    /// assert_eq!(encoding, AssetEncoding::Brotli);
+    /// assert_eq!(extension, "br");
+    ///
+    /// let (encoding, extension) = AssetEncoding::Zstd.default();
+    /// assert_eq!(encoding, AssetEncoding::Zstd);
+    /// assert_eq!(extension, "zst");
+    ///
+    /// let (encoding, extension) = AssetEncoding::Gzip.default();
+    /// assert_eq!(encoding, AssetEncoding::Gzip);
+    /// assert_eq!(extension, "gz");
+    ///
+    /// let (encoding, extension) = AssetEncoding::Deflate.default();
+    /// assert_eq!(encoding, AssetEncoding::Deflate);
+    /// assert_eq!(extension, "zz");
+    /// ```
+    pub fn default(self) -> (AssetEncoding, String) {
+        let file_extension = match self {
+            AssetEncoding::Brotli => "br".to_string(),
+            AssetEncoding::Zstd => "zst".to_string(),
+            AssetEncoding::Gzip => "gz".to_string(),
+            AssetEncoding::Deflate => "zz".to_string(),
+        };
+
+        (self, file_extension)
+    }
+
+    /// Returns an encoding with a custom file extension. This is useful
+    /// when the default file extension assigned by [default](AssetEncoding::default)
+    /// is not desired.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use ic_asset_certification::AssetEncoding;
+    ///
+    /// let (encoding, extension) = AssetEncoding::Brotli.custom("brotli".to_string());
+    ///
+    /// assert_eq!(encoding, AssetEncoding::Brotli);
+    /// assert_eq!(extension, "brotli");
+    /// ```
+    pub fn custom(self, extension: String) -> (AssetEncoding, String) {
```

Comment:
nit: if `default()` above is renamed, then `custom()` should follow the same pattern.